STORM-3957: People page now links to ASF's https://projects.apache.or…

avermeer opened the pull request:
…g/committee.html?storm

rzo1 commented:
@avermeer I think you need to run `bundle exec jekyll build -d content` to update the `content` directory.

rzo1 reviewed (changes requested):
@avermeer I think you need to run `bundle exec jekyll build -d content` to update the `content` directory.

avermeer commented:
> @avermeer I think you need to run `bundle exec jekyll build -d content` to update the `content` directory.

Hello Richard,

Thanks for suggesting me to run `bundle exec jekyll build -d content` ...

... but that leads to huge number of deleted files, so i'm not sure it
is appropriate for the change which I'm trying to deliver on "People"
page:

$ git status
On branch STORM-3957
Your branch is up to date with 'origin/STORM-3957'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        new file:   .jekyll-metadata
        modified:   Gemfile.lock
        modified:   content/Gemfile.lock
        modified:   content/contribute/People.html
        modified:   content/feed.xml
        new file:   content/releases/current
        deleted:
content/releases/current/Acking-framework-implementation.html
        deleted:    content/releases/current/Classpath-handling.html
        deleted:    content/releases/current/Clojure-DSL.html
        deleted:    content/releases/current/ClusterMetrics.html
        deleted:    content/releases/current/Command-line-client.html
        deleted:    content/releases/current/Common-patterns.html
        deleted:    content/releases/current/Concepts.html
        deleted:    content/releases/current/Configuration.html
        deleted:    content/releases/current/Contributing-to-Storm.html
        deleted:    content/releases/current/Creating-a-new-Storm-project.html
        deleted:    content/releases/current/DSLs-and-multilang-adapters.html
        deleted:    content/releases/current/Daemon-Fault-Tolerance.html
        deleted:
content/releases/current/Defining-a-non-jvm-language-dsl-for-storm.html
        deleted:    content/releases/current/Distributed-RPC.html
        deleted:    content/releases/current/Docker-support.html
        deleted:    content/releases/current/Eventlogging.html
        deleted:    content/releases/current/FAQ.html
... (many more lines like this)...

Am I missing something ?

I'm waiting for more inputs before doing more mistakes :)

Thanks,
Alexandre

rzo1 commented:
Looks like the symlink to current is missing :-)

avermeer commented:
> Looks like the symlink to current is missing :-)

Okay so I think have understood the reason for my mistake: I tried to use Windows operating system to work on this change on storm-site.

Looks like symlink support on Windows isn't quite working.

Let me switch to Linux... maybe a word of warning in storm-site's README.md to avoid using Windows for running bundle commands would help the next newbie committer to fall into same pitfall as I did ;)

rzo1 commented:
Another idea would be to use the ASF Jenkins CI to build from a different branch and automatically push the content changes to `asf-site`. So we only need to deal with the `.md` files and the test is a CI job.

avermeer commented:
> Another idea would be to use the ASF Jenkins CI to build from a different branch and automatically push the content changes to `asf-site`. So we only need to deal with the `.md` files and the test is a CI job.

Maybe, but I have no idea how to do this. In the meantime, i'm trying to move on Linux my storm-site workspace. Fighting hard to have proper Ruby version & dependencies to get `gem install jekyll bundler` to work - I'm on a Redhat 8.6 machine and by default it gets Ruby 2.6.0.

After a while, I got Ruby 3.0.0 installed and that seems to make `gem install jekyll bundler` working, but then `bundle install` failed until I ran it using `sudo bundle install` (in spite of the warning messages telling it's not a good idea to run it as root).

I must confess all this stuff for making minor changes to storm-site seems to me like a _gas factory_. I have the feeling that using LaTeX for Storm documentation would have been even simpler (don't take me seriously for this one).

rzo1 commented:
> I must confess all this stuff for making minor changes to storm-site seems to me like a gas factory. I have the feeling that using LaTeX for Storm documentation would have been even simpler (don't take me seriously for this one).

Given that I had the same learning curve as you for my latest changes to `storm-site`, I know (!) that feeling by heart. If I find some time, I can dig into the CI thing (we have something similar in other projects as well).

avermeer commented:
I have git pushed the changes from `bundle exec jekyll build -d content`, can I merge this pull request, or should I wait 2/3 days to leave others time to comment?

rzo1 commented:
I think you are good to go.

avermeer commented:
> I think you are good to go.

Done, and I see that https://storm.apache.org/contribute/People.html is now up to date.
Many thanks for the help!
